packages/api-lib/libs/ingest.js

```diff
@@ -32,6 +33,49 @@ function getChildLinks(node) {
   return links
 }
 
+function getS3ParamsFromUrl(url) {
+  if (!url.startsWith('s3://')) {
+    const msg = `cannot parse bucket/key from ${url}; url must start with "s3://"`
+    logger.error(msg)
+    throw new Error(msg)
+  }
+  const urlWithoutProtocol = url.replace('s3://', '')
+  const allParts = urlWithoutProtocol.split('/')
+  if (allParts.length < 2) {
+    const msg = `cannot parse bucket/key from ${url}; url does not contain bucket and key`
+    logger.error(msg)
+    throw new Error(msg)
+  }
+  const bucket = allParts[0]
+  const remainingParts = allParts.slice(1)
+  const key = remainingParts.join('/')
+  logger.debug(`parsed Bucket: ${bucket} Key: ${key} from ${url}`)
+  return {
+    RequestPayer: process.env.AWS_REQUEST_PAYER,
+    Bucket: bucket,
+    Key: key
+  }
+}
+
+function getS3Object(url) {
+  // from https://github.com/aws/aws-sdk-js/issues/1436
+  return new Promise((resolve, reject) => {
+    const s3 = new AWS.S3()
+    s3.getObject(
+      getS3ParamsFromUrl(url),
```

Comment:
I haven't actually tested this, but it appears that in this case if `getS3ParametersFromUrl` throws an error it won't be caught and execution will stop.  I'd recommend moving the `getS3ParamsFromUrl` out of the `s3.getObject` call and reject the containing promise if an error is thrown.